### PR TITLE
Filter shortest path

### PR DESCRIPTION
**Pull request by PianoRollRepresentation**
This PR adds the UI in the filter menu for calculating the shortest path of #232.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "@cypress/react": "^5.4.0",
     "@material-ui/core": "^4.11.4",
     "@material-ui/icons": "^4.11.2",
+    "@material-ui/lab": "^4.0.0-alpha.58",
```
Comment (CatoLeanTruetschel) [inline frontend/package.json:34]: Why do we need this experimental, unstable library?

File: frontend/src/visualization/filtering/ShortestPathMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import { Box, Button } from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+import ShortestPathNodeSelection from './ShortestPathNodeSelection';
+
+/**
+ * Menu where start and end node for shortest path can be selected
+ */
+export default function ShortestPathMenu(): JSX.Element {
+  const [errorNotSpecified, setErrorNotSpecified] = useState(false);
```
Comment (CatoLeanTruetschel) [inline frontend/src/visualization/filtering/ShortestPathMenu.tsx:10]: The variable name is very cumbersome to read. Can you store the opposite, such that the nameing simplifies to `errorSpecified`?

File: frontend/src/visualization/filtering/ShortestPathMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  const [startNode, setStartNode] = useState<number | null>(null);
+  const [endNode, setEndNode] = useState<number | null>(null);
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
```
Comment (CatoLeanTruetschel) [inline frontend/src/visualization/filtering/ShortestPathMenu.tsx:15]: Why is this needed? This is not an empty function. Is this an artifacts that is remaining from the development process, where this was empty?

File: frontend/src/visualization/filtering/ShortestPathMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      // TODO: compute shortest path. Use startNode and endNode.
+    } else {
+      setErrorNotSpecified(true);
+    }
+  };
+
+  return (
+    <>
+      <Box display="flex" p={1} fontWeight={500} fontSize={16}>
+        Shortest path computation
```
Comment (CatoLeanTruetschel) [inline frontend/src/visualization/filtering/ShortestPathMenu.tsx:28]: Remove computation or name it `Find shortest path`

File: frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import { map } from 'rxjs/operators';
+import useService from '../../dependency-injection/useService';
+import QueryResultStore from '../../stores/QueryResultStore';
+import useObservable from '../../utils/useObservable';
+
+/**
+ * Selection of a start or an end node using a {@link Autocomplete}.
+ */
+export default function ShortestPathNodeSelection(props: {
+  end: 'start' | 'end';
```
Comment (CatoLeanTruetschel) [inline frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx:13]: Rename end property to `tail`

File: frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            {...params}
+            label={end}
+            variant="outlined"
+            required
+          />
+        )}
+        PopperComponent={AutocompletePopper}
+        ListboxProps={{
+          style: {
+            maxHeight: '200px',
```
Comment (CatoLeanTruetschel) [inline frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx:65]: Don't use fix pixel values

File: frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  );
+
+  return (
+    <>
+      <Autocomplete
+        className="StartEndNode"
+        onChange={handleSetNode}
+        options={nodeIds}
+        getOptionLabel={(option) => option.toString()}
+        style={{ width: 200 }}
```
Comment (CatoLeanTruetschel) [inline frontend/src/visualization/filtering/ShortestPathNodeSelection.tsx:51]: Don't use fix pixel values

**Review by CatoLeanTruetschel — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 1 above)
Comment (PianoRollRepresentation) [reply frontend/package.json:34]: Autocomplete is a component of material-ui/lab. Not sure if there's something comparable in material-ui/core (have not find something there). If you know some material-ui alternative let me know!

File: frontend/src/visualization/filtering/ShortestPathMenu.tsx (hunk 3 above)
Comment (PianoRollRepresentation) [reply frontend/src/visualization/filtering/ShortestPathMenu.tsx:15]: yes its an artificat, I will delete that

File: frontend/package.json (hunk 1 above)
Comment (CatoLeanTruetschel) [reply frontend/package.json:34]: I'm not entirely happy with using this experimantal tech, but leave it for now.

File: frontend/src/visualization/filtering/ShortestPathMenu.tsx [hunk 8]
```diff
@@ -7,31 +7,30 @@ import ShortestPathNodeSelection from './ShortestPathNodeSelection';
...
       <Box display="flex" p={1} fontWeight={500} fontSize={16}>
-        Shortest path computation
+        Find shortest path
       </Box>
       <Box display="flex" p={1}>
         <ShortestPathNodeSelection end="start" setNode={setStartNode} />
       </Box>
       <Box display="flex" p={1}>
-        <ShortestPathNodeSelection end="end" setNode={setEndNode} />
+        <ShortestPathNodeSelection end="tail" setNode={setEndNode} />
```
Comment (CatoLeanTruetschel) [inline frontend/src/visualization/filtering/ShortestPathMenu.tsx:33]: I meant to rename to property to tail, not the property value. Such that this would look like
`tail=start` and `tail=start`

**Review by CatoLeanTruetschel — CHANGES_REQUESTED** [review]

**Review by CatoLeanTruetschel — APPROVED** [review]
